control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3178,13 +3173,12 @@ func (r *HostedControlPlaneReconciler) reconcileKubeAPIServer(ctx context.Contex
 			}
 
 			if hyperazureutil.IsAroHCP() {
-				// Reconcile the SecretProviderClass
-				kmsSecretProviderClass := manifests.ManagedAzureSecretProviderClass(config.ManagedAzureKMSSecretProviderClassName, hcp.Namespace)
-				if _, err := createOrUpdate(ctx, r, kmsSecretProviderClass, func() error {
-					secretproviderclass.ReconcileManagedAzureSecretProviderClass(kmsSecretProviderClass, hcp, hcp.Spec.SecretEncryption.KMS.Azure.KMS, true)
-					return nil
+				// Reconcile the Azure KMS Config Secret
```

Comment:
should this be part of hc/internal/platform ReconcileCredentials for consistency with aws? is the cloudNetworkConfigCreds there still needed?
alternative should we make this part of the kms.IKMSProvider interface?

Reply:
cloudNetworkConfigCreds is still needed for CCM/CP so no need to mess with it here.